coderd/coderd.go

```diff
@@ -1333,6 +1333,7 @@ func New(options *Options) *API {
 								r.Get("/tokenconfig", api.tokenConfig)
 								r.Route("/{keyname}", func(r chi.Router) {
 									r.Get("/", api.apiKeyByName)
+									r.Patch("/", api.patchToken)
```

Comment:
Why do we need to update api keys? 

Is this a common practice elsewhere? Can the user just make a new api key with the new scopes? I feel strange allowing someone to escalate a token after it has been created.

Reply:
Is this required for oauth, or has product requested it?

I would prefer not to add this if it is not requested/required.

Reply:
> Is this a common practice elsewhere? Can the user just make a new api key with the new scopes? I feel strange allowing someone to escalate a token after it has been created.

It is a common practice. GitHub, for example, allows you to update a personal access token and attach additional permissions to it or remove permissions. This is not only an escalation but also a de-escalation.

> Is this required for oauth, or has product requested it?
> I would prefer not to add this if it is not requested/required.

It's part of the API key scopes RFC.

Reply:
> It's part of the API key scopes RFC.

Do you have a quote from it? I cannot find information of updating a key's scope/allow_list